pkg/microservice/aslan/core/environment/service/environment.go

```diff
@@ -907,6 +939,44 @@ func UpdateHelmProduct(productName, envName, updateType, username, requestID str
 		log.Errorf("[%s][P:%s] GetProductTemplate error: %v", envName, productName, err)
 		return e.ErrUpdateEnv.AddDesc(e.FindProductTmplErrMsg)
 	}
+
+	for _, svrs := range updateProd.Services {
+		for _, svr := range svrs {
+			for _, rc := range renderSet.ChartInfos {
+				if rc.ServiceName == svr.ServiceName {
+					for _, c := range svr.Containers {
+						image := c.Image
+						mergeYaml, err := helmtool.MergeOverrideValues(rc.ValuesYaml, renderSet.DefaultValues, rc.GetOverrideYaml(), rc.OverrideValues)
+						if err != nil {
+							errMsg := fmt.Sprintf("Failed to MergeOverrideValues for service  %s,error:%s", svr.ServiceName, err)
+							log.Error(errMsg)
+							return e.ErrUpdateEnv.AddDesc(errMsg)
+						}
+						mergedValuesYamlFlattenMap, err := converter.YamlToFlatMap([]byte(mergeYaml))
+						if err != nil {
+							errMsg := fmt.Sprintf("mergeYamlToFlatMap service %s,error:%s", svr.ServiceName, err)
+							log.Error(errMsg)
+							return e.ErrUpdateEnv.AddDesc(errMsg)
+						}
+						imageRepo, ok1 := mergedValuesYamlFlattenMap[c.ImagePath.Repo]
+						imageTag, ok2 := mergedValuesYamlFlattenMap[c.ImagePath.Tag]
+						if ok2 {
+							if ok1 {
+								image = fmt.Sprintf("%s:%s", imageRepo, imageTag)
+							} else {
+								splitImage := strings.Split(image, ":")
+								if len(splitImage) == 2 {
+									image = fmt.Sprintf("%s:%s", splitImage[0], imageTag)
+								}
+							}
+						}
+						c.Image = image
```

Comment:
is it ok to use image in product.service[].containers[].image instead of extracting image from whole values.yaml?